staging/src/k8s.io/client-go/tools/cache/delta_fifo.go

```diff
@@ -310,13 +303,6 @@ func (f *DeltaFIFO) queueActionLocked(actionType DeltaType, obj interface{}) err
 		return KeyError{obj, err}
 	}
 
-	// If object is supposed to be deleted (last event is Deleted),
-	// then we should ignore Sync events, because it would result in
-	// recreation of this object.
```

Comment:
I can't remember the rationale for this comment when it was orignally added, but I think there was some. It would be good to ensure that we won't break something else with this fix...

@liggitt @lavalamp - maybe you remember it?

Reply:
@wojtek-t you added that hunk in #27435 where the related change was:

```
-	} else if _, exists, err := f.knownObjects.GetByKey(id); err == nil && !exists {
-		// Presumably, this was deleted when a relist happened.
-		// Don't provide a second report of the same deletion.
-		// TODO(lavalamp): This may be racy-- we aren't properly locked
-		// with knownObjects.
-		return nil
+	} else {
+		// We only want to skip the "deletion" action if the object doesn't
+		// exist in knownObjects and it doesn't have corresponding item in items.
+		// Note that even if there is a "deletion" action in items, we can ignore it,
+		// because it will be deduped automatically in "queueActionLocked"
+		_, exists, err := f.knownObjects.GetByKey(id)
+		_, itemsExist := f.items[id]
+		if err == nil && !exists && !itemsExist {
+			// Presumably, this was deleted when a relist happened.
+			// Don't provide a second report of the same deletion.
+			// TODO(lavalamp): This may be racy-- we aren't properly locked
+			// with knownObjects.
+			return nil
+		}
```

Reply:
I know I added that, but I can't remember exactly the reasoning (it was couple years ago).

Reply:
@wojtek-t perhaps https://github.com/kubernetes/kubernetes/pull/26167 describes the issue better

Reply:
both specific unit tests added in https://github.com/kubernetes/kubernetes/pull/27435 (TestDeltaFIFO_ResyncNonExisting and TestDeltaFIFO_DeleteExistingNonPropagated) still pass with the proposed fix here, which is positive

Reply:
It looks like the sync-specific handling around not sending Sync events for objects with queued events was added later in https://github.com/kubernetes/kubernetes/pull/32125/files

Reply:
@wojtek-t to expand on the two previous comments, see how in the path which leads to enqueuing of a Sync delta created by Resync() there's the following check: https://github.com/matte21/kubernetes/blob/fix-shared-index-informer-notification-miss/staging/src/k8s.io/client-go/tools/cache/delta_fifo.go#L546-L48 . It ensures the "bad thing" your comment describes does not happen.

Reply:
@matte21 yeah, the comment/issue I referenced may have been valid before Relist and Resync were split apart, but I don't  think it's valid anymore.